src/torrent.cpp

```diff
@@ -2285,26 +2292,26 @@ namespace libtorrent
 #endif
 				)
 			{
-				TORRENT_ASSERT(error.file >= 0);
+				TORRENT_ASSERT(error.file() >= file_index_t(0));
 
 				// skip this file by updating m_checking_piece to the first piece following it
 				file_storage const& st = m_torrent_file->files();
-				std::uint64_t file_size = st.file_size(error.file);
-				int last = st.map_file(error.file, file_size, 0).piece;
+				std::uint64_t file_size = st.file_size(error.file());
+				piece_index_t last = st.map_file(error.file(), file_size, 0).piece;
 				if (m_checking_piece < last)
 				{
-					int diff = last - m_checking_piece;
-					m_num_checked_pieces += diff;
-					m_checking_piece += diff;
+					int diff = static_cast<int>(last) - static_cast<int>(m_checking_piece);
```

Comment:
I'm punting on enriching the types here. I'm thinking that perhaps there could be an implied "distance" unit, which is the result of subtraction and can be used in addition.